e2e-tests/src/substrate_api.py

```diff
@@ -68,22 +181,30 @@ def get_latest_pc_block_number(self):
         return block["header"]["number"]
 
     def get_latest_mc_block_number(self):
-        block = self.cardano_cli.get_block()
-        logger.debug(f"Current main chain block: {block}")
-        return block
+        if self.cardano_cli:
```

Comment:
It is better to move refactoring and improvement of the current e2e tests in a separate PR